packages/opencensus-core/src/stats/view.ts

```diff
@@ -112,10 +123,31 @@ export class BaseView implements View {
    * @param tags The tags to encode
    */
   private encodeTags(tags: Tags): string {
-    const encodedTags = Object.keys(tags).sort().reduce((strTags, tagKey) => {
-      return strTags + `${tagKey}:${tags[tagKey]},`;
-    }, '');
-    return `{${encodedTags}}`;
+    return Object.keys(tags)
+        .sort()
+        .map(tagKey => {
+          return tagKey + UNIT_SEPARATOR + tags[tagKey];
+        })
+        .join(RECORD_SEPARATOR);
+  }
+
+  /**
+   * Checks if tag keys and values have only printable characters.
+   * @param tags The tags to be checked
+   */
+  private invalidTags(tags: Tags): boolean {
+    return Object.keys(tags).some(tagKey => {
+      return this.invalidString(tagKey) || this.invalidString(tags[tagKey]);
+    });
+  }
+
+  /**
+   * Checks if a string has only printable characters.
+   * @param str The string to be checked
+   */
+  private invalidString(str: string): boolean {
+    return RegExp(`[${LOW_LIMIT_CHAR}-${HIGH_LIMIT_CHAR}]`).test(str) ||
```

Comment:
Nit: I would suggest creating this `RegExp` in the file scope or as a static field so it doesn't have to get constructed on every call to `invalidString`. Its value is static anyway. (Same for next line.)

Also, would `/[^\u0020-\u007e]/` work as a regex to encompass both?

Reply:
I opted for `RegExp()` for readability when passing `LOW_LIMIT_CHAR` and `HIGH_LIMIT_CHAR` but hadn't considered the performance issue. I think it is way better with `/[^\u0020-\u007e]/`. Thanks again for the tips!